javalib/src/main/scala/java/math/MathContext.scala

```diff
@@ -0,0 +1,206 @@
+   * rounding.
+   */
+  val DECIMAL64 = MathContext(16, RoundingMode.HALF_EVEN)
+
+  /**
+   * A {@code MathContext} for unlimited precision with
+   * {@link RoundingMode#HALF_UP} rounding.
+   */
+  val UNLIMITED = MathContext(0, RoundingMode.HALF_UP)
+
+  private def getArgs(s: String): (Int, RoundingMode) = {
+    checkNotNull(s, "null string")
+    val precisionLength: Int = "precision=".length
+    val roundingModeLength: Int = "roundingMode=".length
+    val spaceIndex: Int = s.indexOf(' ', precisionLength)
+
+    if (!s.startsWith("precision=") || spaceIndex == -1) {
+      throw invalidMathContext("Missing precision", s)
+    }
+    val precisionString = s.substring(precisionLength, spaceIndex)
+
+    val precision = try {
+        java.lang.Integer.parseInt(precisionString)
+      }
+      catch {
+        case _: Throwable => throw invalidMathContext("Bad precision", s)
+      }
+
+    var roundingModeStart: Int = spaceIndex + 1
+    if (!s.regionMatches(roundingModeStart, "roundingMode=", 0, roundingModeLength)) {
+      throw invalidMathContext("Missing rounding mode", s)
+    }
+    roundingModeStart += roundingModeLength
+    val roundingMode: RoundingMode = RoundingMode.valueOf(s.substring(roundingModeStart))
+
+    (precision, roundingMode)
+  }
+
+  private def invalidMathContext(reason: String, s: String): IllegalArgumentException = {
+    throw new IllegalArgumentException(reason + ": " + s)
+  }
+
+  private def checkNotNull(reference: AnyRef, errorMessage: AnyRef): Unit =
+    if (reference == null)
+      throw new NullPointerException(String.valueOf(errorMessage))
+
+}
+
+/**
+ *  Immutable objects describing settings such as rounding mode and digit
+ *  precision for the numerical operations provided by class {@link BigDecimal}.
+ *
+ *  Constructs a new {@code MathContext} with the specified precision and
+ *  with the specified rounding mode. If the precision passed is zero, then
+ *  this implies that the computations have to be performed exact, the
+ *  rounding mode in this case is irrelevant.
+ *
+ *  @param precision
+ *            the precision for the new {@code MathContext}.
+ *  @param roundingMode
+ *            the rounding mode for the new {@code MathContext}.
+ */
+case class MathContext(precision: Int, roundingMode: RoundingMode) extends Serializable {
+
+  checkValid()
+
+  def getPrecision: Int = precision
+
+  def getRoundingMode: RoundingMode = roundingMode
+
+  /**
+   * Constructs a new {@code MathContext} with the specified precision and
+   * with the rounding mode {@link RoundingMode#HALF_UP HALF_UP}. If the
+   * precision passed is zero, then this implies that the computations have to
+   * be performed exact, the rounding mode in this case is irrelevant.
+   *
+   * @param precision
+   *            the precision for the new {@code MathContext}.
+   */
+  def this(precision: Int) {
+    this(precision, RoundingMode.HALF_UP)
+  }
+
+  private def this(args:(Int, RoundingMode)) {
```

Comment:
Missing space after `:`
Clever trick, btw, the private constructor to give an intermediate name to the tuple :-p